src/librustdoc/json/conversions.rs

```diff
@@ -634,42 +634,28 @@ impl FromWithTcx<clean::Enum> for Enum {
         Enum {
             generics: generics.into_tcx(tcx),
             variants_stripped,
-            variants: ids(variants, tcx),
+            variants: ids(&variants, tcx),
             impls: Vec::new(), // Added in JsonRenderer::item
         }
     }
 }
 
-impl FromWithTcx<clean::VariantStruct> for Struct {
-    fn from_tcx(struct_: clean::VariantStruct, tcx: TyCtxt<'_>) -> Self {
-        let fields_stripped = struct_.has_stripped_entries();
-        let clean::VariantStruct { struct_type, fields } = struct_;
-        Struct {
-            struct_type: from_ctor_kind(struct_type),
-            generics: Generics { params: vec![], where_predicates: vec![] },
-            fields_stripped,
-            fields: ids(fields, tcx),
-            impls: Vec::new(),
-        }
-    }
-}
-
 impl FromWithTcx<clean::Variant> for Variant {
     fn from_tcx(variant: clean::Variant, tcx: TyCtxt<'_>) -> Self {
         use clean::Variant::*;
+
         match variant {
-            CLike => Variant::Plain,
-            Tuple(fields) => Variant::Tuple(
-                fields
-                    .into_iter()
-                    .filter_map(|f| match *f.kind {
-                        clean::StructFieldItem(ty) => Some(ty.into_tcx(tcx)),
-                        clean::StrippedItem(_) => None,
-                        _ => unreachable!(),
-                    })
-                    .collect(),
-            ),
-            Struct(s) => Variant::Struct(ids(s.fields, tcx)),
+            CLike => Variant { kind: StructKind::Unit, fields: Vec::new(), fields_stripped: false },
+            Tuple(fields) => Variant {
+                kind: StructKind::Tuple,
+                fields: ids(&fields, tcx),
+                fields_stripped: fields.iter().any(|i| i.is_stripped()),
```

Comment:
I've been experimenting with this code (see https://github.com/Enselic/cargo-public-api/pull/99) and have the following question/concern: What is the value of having a bool to specify if some field is stripped? Wouldn't it be better to remove `fields_stripped` and instead add info on a per-field level if it is stripped or not? Later today I think I will have time to develop a concrete proposal on how it could look.

Reply:
I think the idea is to not show stripped fields and just say more fields exist but they're not part of the public API.

Reply:
I think it would be good to have higher fidelity than "more fields exist" and also be able to easily say which fields have been stripped. In other words, be on par with rustdoc HTML output, which looks like this:

```rust
pub enum EnumWithStrippedTupleVariants {
    Double(bool, bool),
    DoubleFirstHidden(_, bool),
    DoubleSecondHidden(bool, _),
}
```

Reply:
Some options for this:

1: Also give the number of fields. The relative position of fields can be determined by the number in the name for tuple, and not at all for structs.
2: Instead of having a `Vec<Id>`, have a Vec<Option<Id>>`, and use `None` for stripped fields.

Reply:
2 sounds good! I haven't had time to experiment but that should work well.

Not a fan of 1 because names having special meaning seems too fragile for an API.